### PR TITLE
PD-864 Wait for search results before sending Products Searched

### PR DESCRIPTION
**Pull request by edwardupton**

### CODE REVIEW

File: src/segmentTracker/helpers/sendEventsWithPageview.ts [hunk 1]
```diff
@@ -19,9 +19,23 @@ export const sendEventsWithPageview = (pathname: string): void => {
 		trackEvent('Registration Viewed', {});
 	}
 	if (pathname === '/search') {
+		trackSearch();
+	}
+};
+
+function trackSearch(secondAttempt?: boolean): void {
```
Comment (lyykfi) [inline src/segmentTracker/helpers/sendEventsWithPageview.ts:26]: could you add a comment please?

File: src/segmentTracker/helpers/sendEventsWithPageview.ts [hunk 2]
```diff
@@ -19,9 +19,23 @@ export const sendEventsWithPageview = (pathname: string): void => {
...
+	const results = getProductLinks(document, LittledataLayer.productListLinksHaveImages).length;
+
+	if (results == 0 || secondAttempt) {
```
Comment (lyykfi) [inline src/segmentTracker/helpers/sendEventsWithPageview.ts:29]: can be just !results || secondAttempt

File: src/segmentTracker/helpers/sendEventsWithPageview.ts [hunk 3]
```diff
@@ -19,9 +19,23 @@ export const sendEventsWithPageview = (pathname: string): void => {
...
 			query: document.location.search.replace('?q=', '').replace('+', ' '),
-			results: getProductLinks(document, LittledataLayer.productListLinksHaveImages).length,
+			results,
 		});
+
+		return;
 	}
-};
+	// check again in case the products list took a long time to load
+	window.setTimeout(() => {
```
Comment (lyykfi) [inline src/segmentTracker/helpers/sendEventsWithPageview.ts:38]: should we try only twice or maybe many times?

**Comment by edwardupton** [comment]
Added comment.

I think just twice, because if there really are no products we do want to
track the event with zero products

**Review by lyykfi — APPROVED** [review]

**Pull request merged**